xdsl/transforms/printf_to_putchar.py

```diff
@@ -215,38 +214,28 @@ class PrintfToPutcharPass(ModulePass):
     # lower to func.call
     def apply(self, ctx: MLContext, op: ModuleOp) -> None:
         # Check if there are any printints
-        contains_printint = False
-        for op_in_module in op.walk():
-            if isinstance(op_in_module, PrintIntOp):
-                contains_printint = True
-                # One printint encountered is enough
-                break
+        contains_printint = any(
+            isinstance(op_in_module, PrintIntOp) for op_in_module in op.walk()
+        )
         if contains_printint:
```

Comment:
```suggestion
        if any(
            isinstance(op_in_module, PrintIntOp)
            for op_in_module in op.walk()
        ):
```

Reply:
Can combine if only use once for if condition.

Reply:
Again, I think it is more readable if it is explicitly stated